solutions/gold/usaco-384.mdx

```diff
@@ -0,0 +1,114 @@
+---
+id: usaco-384
+source: USACO Gold 2014 January
+title: Ski Course Rating
+author: Chongtian Ma
+---
+
+[Official Analysis (C++)](http://www.usaco.org/current/data/sol_skilevel.html)
+
+## Implementation
+
+<LanguageSection>
+<CPPSection>
+
+Interestingly, the following code passes all test cases but won't pass if $T = 1$; you'd have to make a separate edge case for that.
+
+**Time Complexity:** $\mathcal{O}(N^2 \log N)$
+
+```cpp
+#include <bits/stdc++.h>
+using namespace std;
+using ll = long long;
+
+struct Edge {
+	int u, v, w;
+};
+
+struct DSU {
+	vector<int> p, sz, cnt_start;
+	// cnt_start stores unprocessed start nodes in a component
+	DSU(int n) {
+		p.resize(n);
+		sz.resize(n, 1);
+		cnt_start.resize(n);
+		iota(p.begin(), p.end(), 0);
+	}
+
+	int get(int x) {
+		if (p[x] != x) { p[x] = get(p[x]); }
+		return p[x];
+	}
+
+	void unite(Edge e) {
+		int a = get(e.u);
+		int b = get(e.v);
+		if (sz[a] < sz[b]) swap(a, b);
+		if (a != b) {
+			p[b] = a;
+			sz[a] += sz[b];
+			cnt_start[a] += cnt_start[b];
+		}
+	}
+};
+
+int main() {
+	ifstream read("skilevel.in");
+	int n, m, t;
+	read >> n >> m >> t;
+	vector<vector<int>> grid(n, vector<int>(m));
+	for (int i = 0; i < n; i++) {
+		for (int j = 0; j < m; j++) { read >> grid[i][j]; }
+	}
+	vector<vector<int>> is_start(n, vector<int>(m));
+	for (int i = 0; i < n; i++) {
+		for (int j = 0; j < m; j++) { read >> is_start[i][j]; }
+	}
+
+	auto to_int = [&](int r, int c) -> { return r * m + c; };
```

Comment:
@SansPapyrus683 whyd you remove the `int` now it compile errors 
this aint java

Reply:
oh ic
you used `auto`
sorry bout that, change it back please